File: Data Structures & Algorithms/duplicate-integer/submission-6.go
// brute force, time: O(n^2), space: O(1)

func hasDuplicate(nums []int) bool {
    for i := 0; i < len(nums); i++ {
        for j := i + 1; j < len(nums); j ++ {
            if nums[i] == nums[j]{
                return true
            }
        } 
    }
    return false
}

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-6.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        # dp, time: O(n), space: O(1)

        p2, p1 = 0, 0
        for c in cost:
            p2, p1 = p1, c + (p2 if p2 < p1 else p1)
        return p2 if p2 < p1 else p1